libc-bottom-half/headers/public/__header_sys_socket.h

```diff
@@ -1,19 +1,57 @@
 #ifndef __wasilibc___header_sys_socket_h
 #define __wasilibc___header_sys_socket_h
 
+#include <__wasi_snapshot.h>
 #include <__struct_msghdr.h>
 #include <__struct_sockaddr.h>
 #include <__struct_sockaddr_storage.h>
 
 #include <wasi/api.h>
 
-#define SHUT_RD __WASI_SDFLAGS_RD
-#define SHUT_WR __WASI_SDFLAGS_WR
-#define SHUT_RDWR (SHUT_RD | SHUT_WR)
+#ifdef __wasilibc_use_preview2
+#define MSG_DONTWAIT  0x0040
+#define MSG_NOSIGNAL  0x4000
+#define MSG_PEEK      0x0002
+#define MSG_WAITALL   0x0100
+#define MSG_TRUNC     0x0020
+
+#define SOL_IP     0
+#define SOL_TCP    6
+#define SOL_UDP    17
+#define SOL_IPV6   41
+
+#define SOMAXCONN 128
+
+#define SO_ERROR 4
+#define SO_SNDBUF 7
+#define SO_RCVBUF 8
+#define SO_KEEPALIVE 9
+#define SO_ACCEPTCONN 30
+#define SO_PROTOCOL 38
+#define SO_DOMAIN 39
 
+#if __LONG_MAX == 0x7fffffff
```

Comment:
why?

Reply:
This was taken verbatim from `libc-top-half/musl/include/sys/socket.h` where it lives within a
`#ifdef __wasilibc_unmodified_upstream /* Use alternate WASI libc headers */` 
block.

Reply:
I'm not too familiar with this codebase, but I assumed `#if` was probably there for a reason. If not, I suppose we can just pick one of the definitions.

Reply:
i guess it's for something historical in linux.
no need to mirror it to wasi.